pkg/operator/controller/certificate-publisher/controller.go

```diff
@@ -86,9 +86,11 @@ func New(mgr manager.Manager, operatorNamespace, operandNamespace string) (runti
 	}
 
 	if err := c.Watch(&source.Kind{Type: &operatorv1.IngressController{}}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
-		CreateFunc:  func(e event.CreateEvent) bool { return reconciler.hasSecret(e.Meta, e.Object) },
-		DeleteFunc:  func(e event.DeleteEvent) bool { return reconciler.hasSecret(e.Meta, e.Object) },
-		UpdateFunc:  func(e event.UpdateEvent) bool { return reconciler.secretChanged(e.ObjectOld, e.ObjectNew) },
+		CreateFunc: func(e event.CreateEvent) bool { return reconciler.hasSecret(e.Meta, e.Object) },
+		DeleteFunc: func(e event.DeleteEvent) bool { return reconciler.hasSecret(e.Meta, e.Object) },
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return reconciler.hasSecret(e.MetaNew, e.ObjectNew) && reconciler.secretChanged(e.ObjectOld, e.ObjectNew)
```

Comment:
What happens if ingresscontroller A and ingresscontroller B each has a valid secret reference, then A is updated with a broken reference, and then B is updated to a new but still valid reference?  Doesn't that break A?  

Looks like your intention is to prevent updates to the global secret if any ingresscontroller has an invalid secret reference, right?  If so, it probably would work better to check for invalid references in `ensureRouterCertsGlobalSecret`, `desiredRouterCertsGlobalSecret`, or `Reconcile`.  

Maybe though it would be better not to make an invalid reference in ingresscontroller A block updates related to ingresscontroller B.  Instead, we could to change `ensureRouterCertsGlobalSecret` to get the current list (if it exists) and pass it to `desiredRouterCertsGlobalSecret`, and change `desiredRouterCertsGlobalSecret` to use the current list to fill in the PEM data for ingresscontrollers that don't have secrets.

Reply:
See latest changes. I modified `desiredRouterCertsGlobalSecret` along the lines of your final suggestion.